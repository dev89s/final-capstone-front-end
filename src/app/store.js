import { configureStore } from '@reduxjs/toolkit';

import dataReducer from '../components/Home/createSlice';
import userReducer from './user/userSlice';
import roomReducer from './room/roomSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    dataSlice: dataReducer,
    room: roomReducer,
  },
});

export default store;
