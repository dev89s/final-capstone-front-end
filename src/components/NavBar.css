.nav {
  height: 90vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  list-style: none;
  padding: 2rem 2rem 0 2rem;
}

.logo {
  align-self: center;
}

.logo h1 {
  font-size: 1.5rem;
  padding-top: 2rem;
  font-weight: 800;
  background: -webkit-linear-gradient(45deg, greenyellow, red 80%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.links {
  padding-top: 10rem;
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
}

.links > li {
  padding: 1.19rem;
  white-space: nowrap;
}

.social {
  display: flex;
  gap: 0.5rem;
  list-style: none;
  padding: 0;
}

.active {
  font-weight: bold;
  background: greenyellow;
  color: white !important;
  width: 100%;
}

.link {
  text-decoration: none;
  color: black;
}

.mobile {
  display: none;
  z-index: 10;
}

.ham-menu {
  background: none;
  border: none;
}

.social-logo {
  height: 2rem;
  width: 2rem;
}

.copyright {
  text-align: center;
  font-size: 0.7rem;
  font-weight: bold;
}

@media (max-width: 600px) {
  .nav {
    position: fixed;
    background-color: white;
    opacity: 0.95;
    z-index: 99;
  }

  .mobile {
    position: fixed;
    top: 2rem;
    left: 2rem;
    display: inline;
    z-index: 100;
  }

  .open {
    display: none;
  }

  .ham-menu {
    float: right;
  }

  .close-btn {
    position: absolute;
    top: 0;
    left: 8rem;
    font-weight: bold;
    font-size: 1.5rem;
    margin-right: auto;
    z-index: 100;
  }
}
