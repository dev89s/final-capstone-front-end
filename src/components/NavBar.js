import React, { useEffect, useState } from 'react';
import { Link, useLocation } from 'react-router-dom';
import './NavBar.css';

const NavBar = () => {
  const params = useLocation();
  const [open, setOpen] = useState(true);

  const logOut = () => {
    localStorage.removeItem('username');
  };

  useEffect(() => {
    setOpen(true);
  }, [params]);

  return (
    <>
      <div className="mobile">
        <button
          className="ham-menu"
          type="button"
          onClick={() => setOpen(!open)}
        >
          {open ? (
            <svg viewBox="0 0 100 80" width="40" height="40" fill="black">
              <rect width="100" height="10" />
              <rect y="30" width="100" height="10" />
              <rect y="60" width="100" height="10" />
            </svg>
          ) : (
            <p className="close-btn">X</p>
          )}
        </button>
      </div>
      <ul className={`nav ${open && 'open'}`}>
        <li className="logo">
          <div className="content">
            <h2>SummerWave</h2>
            <h2>SummerWave</h2>
          </div>
        </li>
        <span className="links">
          <li className={`${params.pathname === '/' && 'active'}`}>
            <Link className="link" to="/">
              Room
            </Link>
          </li>
          <li
            className={`${
              params.pathname.split('/')[1] === 'reserve' && 'active'
            }`}
          >
            <Link className="link" to="/reserve">
              Reserve
            </Link>
          </li>
          <li
            className={`${
              params.pathname.split('/')[1] === 'my_reservations' && 'active'
            }`}
          >
            <Link className="link" to="/my_reservations">
              My Reservations
            </Link>
          </li>
          <li
            className={`${
              params.pathname.split('/')[1] === 'add_room' && 'active'
            }`}
          >
            <Link className="link" to="/add_room">
              Add Room
            </Link>
          </li>
          <li
            className={`${
              params.pathname.split('/')[1] === 'delete_room' && 'active'
            }`}
          >
            <Link className="link" to="/delete_room">
              Delete Room
            </Link>
          </li>
          <li>
            <Link className="link" onClick={logOut} to="/login">
              Log Out
            </Link>
          </li>
        </span>
        <li>
          <ul className="social">
            <li>
              <Link to="/#">
                <img className="social-logo" src="/images/x.webp" alt="x" />
              </Link>
            </li>
            <li>
              <Link to="/#">
                <img
                  className="social-logo"
                  src="/images/facebook.webp"
                  alt="facebook"
                />
              </Link>
            </li>
            <li>
              <Link to="/#">
                <img
                  className="social-logo"
                  src="/images/instagram.webp"
                  alt="instagram"
                />
              </Link>
            </li>
            <li>
              <Link to="/#">
                <img
                  className="social-logo"
                  src="/images/pintrest.png"
                  alt="pintrest"
                />
              </Link>
            </li>
          </ul>
        </li>
        <li className="copyright">
          <p>Â© Microverse 2023</p>
        </li>
      </ul>
    </>
  );
};

export default NavBar;
