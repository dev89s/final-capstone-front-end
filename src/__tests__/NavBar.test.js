import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import NavBar from '../components/NavBar';
import store from '../app/store';

describe('Should render the Navbar page', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <NavBar />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have the rooms link text', () => {
    render(
      <Provider store={store}>
        <NavBar />
      </Provider>,
      { wrapper: BrowserRouter },
    );
    const text = screen.getByText('Room');
    expect(text).toBeInTheDocument();
  });

  it('should have the copywrite text', () => {
    render(
      <Provider store={store}>
        <NavBar />
      </Provider>,
      { wrapper: BrowserRouter },
    );
    const text = screen.getByText('Â© Microverse 2023');
    expect(text).toBeInTheDocument();
  });
});
