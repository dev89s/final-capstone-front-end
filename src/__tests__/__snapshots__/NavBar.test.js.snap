// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Navbar page should render correctly 1`] = `
Array [
  <div
    className="mobile"
  >
    <button
      className="ham-menu"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="black"
        height="40"
        viewBox="0 0 100 80"
        width="40"
      >
        <rect
          height="10"
          width="100"
        />
        <rect
          height="10"
          width="100"
          y="30"
        />
        <rect
          height="10"
          width="100"
          y="60"
        />
      </svg>
    </button>
  </div>,
  <ul
    className="nav open"
  >
    <li
      className="logo"
    >
      <h1>
        SummerWave
      </h1>
    </li>
    <span
      className="links"
    >
      <li
        className="active"
      >
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Room
        </a>
      </li>
      <li
        className="false"
      >
        <a
          className="link"
          href="/reserve"
          onClick={[Function]}
        >
          Reserve
        </a>
      </li>
      <li
        className="false"
      >
        <a
          className="link"
          href="/my_reservations"
          onClick={[Function]}
        >
          My Reservations
        </a>
      </li>
      <li
        className="false"
      >
        <a
          className="link"
          href="/add_room"
          onClick={[Function]}
        >
          Add Room
        </a>
      </li>
      <li
        className="false"
      >
        <a
          className="link"
          href="/delete_room"
          onClick={[Function]}
        >
          Delete Room
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/login"
          onClick={[Function]}
        >
          Log Out
        </a>
      </li>
    </span>
    <li>
      <ul
        className="social"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="x"
              className="social-logo"
              src="/images/x.webp"
            />
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="facebook"
              className="social-logo"
              src="/images/facebook.webp"
            />
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="instagram"
              className="social-logo"
              src="/images/instagram.webp"
            />
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="pintrest"
              className="social-logo"
              src="/images/pintrest.png"
            />
          </a>
        </li>
      </ul>
    </li>
    <li
      className="copyright"
    >
      <p>
        Â© Microverse 2023
      </p>
    </li>
  </ul>,
]
`;
